From c377a6f3ad3d9bed4ce7e873e8e9ec6b1650c57d Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Tue, 28 May 2019 12:40:17 +0900
Subject: [PATCH 1/2] journal: do not trigger assertion when
 journal_file_close() get NULL
Origin: upstream, https://github.com/systemd/systemd/pull/12679/commits/c377a6f3ad3d9bed4ce7e873e8e9ec6b1650c57d
Bug: https://github.com/systemd/systemd/issues/12400
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1837914

We generally expect destructors to not complain if a NULL argument is passed.

Closes #12400.
---
 src/journal/journal-file.c    | 3 ++-
 src/journal/journald-server.c | 7 ++-----
 2 files changed, 4 insertions(+), 6 deletions(-)

--- a/src/journal/journal-file.c
+++ b/src/journal/journal-file.c
@@ -348,7 +348,8 @@ bool journal_file_is_offlining(JournalFi
 }
 
 JournalFile* journal_file_close(JournalFile *f) {
-        assert(f);
+        if (!f)
+                return NULL;
 
 #if HAVE_GCRYPT
         /* Write the final tag */
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -1923,11 +1923,8 @@ void server_done(Server *s) {
 
         client_context_flush_all(s);
 
-        if (s->system_journal)
-                (void) journal_file_close(s->system_journal);
-
-        if (s->runtime_journal)
-                (void) journal_file_close(s->runtime_journal);
+        (void) journal_file_close(s->system_journal);
+        (void) journal_file_close(s->runtime_journal);
 
         ordered_hashmap_free_with_destructor(s->user_journals, journal_file_close);
 
